go/adbc/drivermgr/wrapper_sqlite_test.go

```diff
@@ -334,6 +334,85 @@ func (dm *DriverMgrSuite) TestGetObjectsTableType() {
 	dm.False(rdr.Next())
 }
 
+func (dm *DriverMgrSuite) TestGetTableSchema() {
+	schema, err := dm.conn.GetTableSchema(dm.ctx, nil, nil, "test_table")
+	dm.NoError(err)
+
+	expSchema := arrow.NewSchema(
+		[]arrow.Field{
+			{Name: "id", Type: arrow.PrimitiveTypes.Int64, Nullable: true},
+			{Name: "name", Type: arrow.PrimitiveTypes.Int64, Nullable: true}, // Should be arrow.BinaryTypes.String
```

Comment:
I would have expected the SQLite type `TEXT` to map to `arrow.BinaryTypes.String`. I've confirmed via the SQLite driver separately (see screenshot) that this is in fact the behavior of the current driver implementation, so this test actually does indicate the correct behavior for the CGO wrapper.

![image](https://github.com/apache/arrow-adbc/assets/33523178/7085a47f-a421-448c-b41d-d86bfcbedbdf)

Reply:
SQLite doesn't have types. So the only thing the driver can do is try to read some rows and infer an appropriate column type.

Reply:
Thanks @lidavidm, I didn't realize that. This now works as expected after inserting test data.